### PR TITLE
feat(appconnect): Introduce an endpoint to trigger refresh of builds

### PR DESCRIPTION
**Pull request by flub**
Normally the builds are refreshed once an hour, however we're adding a
way to trigger this manually.  This endpoint is still severely
ratelimited.

NATIVE-139

### CODE REVIEW

**Review by loewenheim — APPROVED** [review]

**Review by AniketDas-Tekky — CHANGES_REQUESTED** [review]
There should be unit tests to ensure correct API responses.

**Review by wedamija — COMMENTED** [review]
Probably also worth adding this endpoint to codeowners

**Review by AniketDas-Tekky — COMMENTED** [review]
Who's going to call this API? What is their authentication going to be? Could we restrict which authentication class is used?

**Comment by flub** [comment]
> Who's going to call this API? What is their authentication going to be? Could we restrict which authentication class is used?

I guess one could wonder why this is `StrictProjectPermission` vs `ProjectSettingPermission`, this what you're alluding to?  I don't know the answer to this though as I was not involved with this initial choice.  The entire `projects/{org_slug}/{proj_slug}/appstoreconnect/` hierarchy uses the same permissions set, as it should.  For this PR changing that permission is huge scope creep as it involves changing permissions of existing APIs etc.  If you really believe this should be re-evaluated I think that's a separate issue to resolve.

File: src/sentry/api/endpoints/project_app_store_connect_credentials.py
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_app_store_connect_credentials.py:28]: good catch!

**Review by priscilawebdev — APPROVED** [review]
tested locally and everything seems to work fine

**Comment by flub** [comment]
(FWIW #33576 did get merged into this PR by accident, please ignore the changes to the frontend.  This was supposed to be merged into master once this PR landed there.)

**Review by AniketDas-Tekky — APPROVED** [review]

**Comment by AniketDas-Tekky** [comment]
> > Who's going to call this API? What is their authentication going to be? Could we restrict which authentication class is used?
> 
> I guess one could wonder why this is `StrictProjectPermission` vs `ProjectSettingPermission`, this what you're alluding to? I don't know the answer to this though as I was not involved with this initial choice. The entire `projects/{org_slug}/{proj_slug}/appstoreconnect/` hierarchy uses the same permissions set, as it should. For this PR changing that permission is huge scope creep as it involves changing permissions of existing APIs etc. If you really believe this should be re-evaluated I think that's a separate issue to resolve.

What I meant was the `authentication_classes` not `permissions`. By default we allow Session, Basic, and Token authentication. Meaning anyone with a token could hit this endpoint if they wanted. We could restrict it to Session only. while it won't stop crafty users for hitting the endpoint, it should reduce it/make it harder.

I'm just trying to see if there's a way to avoid being forced to support a private API

**Comment by flub** [comment]
> What I meant was the `authentication_classes` not `permissions`. By default we allow Session, Basic, and Token authentication. Meaning anyone with a token could hit this endpoint if they wanted. We could restrict it to Session only. while it won't stop crafty users for hitting the endpoint, it should reduce it/make it harder.

Ah, I see.  To be honest having "Token" made testing this locally a lot easier :)

> I'm just trying to see if there's a way to avoid being forced to support a private API

More seriously, I actually think all the `.../appstoreconnect/*` APIs should be made public.  There really is nothing that should stop a customer from building some automation for this, maybe they want to regularly expire and refresh these credentials.  It is only private right now out of, err, no one gave us time to build it publicly and this wasn't a thing yet when it was first created.

**Pull request merged**
